compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -554,7 +554,8 @@ createProjectPackageDb opts thisSdkVer deps0 dataDeps = do
         forM allDalfs $ \(name, dalf) -> do
             (pkgId, package) <-
                 either (fail . DA.Pretty.renderPretty) pure $
-                Archive.decodeArchive dalf
+                -- FIXME(MH): This keeps the old behaviour but seems wrong to me.
```

Comment:
No need to rewrite self references here, the dalf won't be linked or typechecked and for code generation it's easier if you just have the PRSelf references and translate those to the unit it.